### PR TITLE
Fix broken links with `--document-private-items` in the standard library

### LINKED ISSUE #81037: Some internal rustdoc links broke
https://github.com/rust-lang/rust/blob/dcf622eb70aebe16d40c5f88fa2a41fa7019541c/library/alloc/src/vec/spec_from_iter_nested.rs#L8

```
warning: unresolved link to `SpecFromIter`
 --> library/alloc/src/vec/spec_from_iter_nested.rs:8:10
  |
8 | /// see [`SpecFromIter`] for details.
  |          ^^^^^^^^^^^^^^ no item named `SpecFromIter` in scope
  |
  = note: `#[warn(broken_intra_doc_links)]` on by default
  = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

And in [stdarch submodule](https://github.com/rust-lang/stdarch/):
```
warning: unresolved link to `B,W`
   --> library/std/src/../../stdarch/crates/std_detect/src/detect/macros.rs:57:19
    |
57  |                   $(#[$feature_comment])*
    |                     ^^^^^^^^^^^^^^^^^^^
    | 
   ::: library/std/src/../../stdarch/crates/std_detect/src/detect/arch/x86.rs:18:1
    |
18  | / features! {
19  | |     @TARGET: x86;
20  | |     @MACRO_NAME: is_x86_feature_detected;
21  | |     @MACRO_ATTRS:
...   |
195 | |     /// RTM, Intel (Restricted Transactional Memory)
196 | | }
    | |_- in this macro invocation
    |
    = note: `#[warn(broken_intra_doc_links)]` on by default
    = note: the link appears in this line:
            
            AVX-512 BITALG (Support for VPOPCNT[B,W] and VPSHUFBITQMB)
                                                ^^^
    = note: no item named `B,W` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

@rustbot label: T-rustdoc C-bug E-easy

### PR DESCRIPTION
**Pull request by CPerezz**
As it was suggested in #81037 `SpecFromIter` is not
in the scope and therefore we get a warning when we try to
do document private intems in `rust/library/alloc/`.

This addresses #81037 by adding the trait in the scope as @jyn514 
suggested and also adding an `allow(unused_imports)` flag so that
the compiler does not complain, Since the trait is not used
per se in the code, it's just needed to have properly documented
docs.

### CODE REVIEW

File: library/alloc/src/vec/spec_from_iter_nested.rs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 use core::iter::TrustedLen;
 use core::ptr::{self};
 
+#[allow(unused_imports)]
+use super::SpecFromIter;
```
Comment (lzutao) [inline library/alloc/src/vec/spec_from_iter_nested.rs:5]: The more expected fix is to change this line
https://github.com/rust-lang/rust/blob/dcf622eb70aebe16d40c5f88fa2a41fa7019541c/library/alloc/src/vec/spec_from_iter_nested.rs#L8
to 
```
/// see [`SpecFromIter`](super::SpecFromIter) for details. 
```

Comment (CPerezz) [reply library/alloc/src/vec/spec_from_iter_nested.rs:5]: My bad. Thanks for the help @lzutao

**Comment by jyn514** [comment]
@bors r+ rollup

I edited the description to remove 'fixes' because stdarch still gives warnings.

**Comment by CPerezz** [comment]
> @bors r+ rollup
> 
> I edited the description to remove 'fixes' because stdarch still gives warnings.

I already made a PR to `stdarch` and got merged to solve them. So should be cleared in the next release.

**Pull request merged**